irods_http_client/zone_operations.py

```diff
@@ -0,0 +1,252 @@
+import requests
+import json
+
+class Zones:
+    def __init__(self, url_base: str):
+        """
+        Initializes Zones with a base url. 
+        Token is set to None initially, and updated when setToken() is called in irodsClient.
+        """
+        self.url_base = url_base
+        self.token = None
+    
+
+    def add(self, name: str, connection_info: str='', comment: str=''):
+        """
+        Adds a remote zone to the local zone. Requires rodsadmin privileges.
+
+        Parameters
+        - name: The name of the zone to be added.
+        - connection_info (optional): The host and port to connect to. If included, must be in the format <host>:<port>
+        - comment (optional): The comment to attach to the zone.
+
+        Returns
+        - A dict containing the HTTP status code and iRODS response.
+        - The iRODS response is only valid if no error occurred during HTTP communication.
+        """
+        if (self.token == None):
+            raise RuntimeError('No token set. Use setToken() to set the auth token to be used')
+        if (not isinstance(name, str)):
+            raise TypeError('name must be a string')
+        if (not isinstance(connection_info, str)):
+            raise TypeError('connection_info must be a string')
+        if (not isinstance(comment, str)):
+            raise TypeError('comment must be a string')
+
+        headers = {
+            'Authorization': 'Bearer ' + self.token,
```

Comment:
This can also be stored as a class member.

Please open an issue for that and we'll handle it in a separate PR.

Reply:
Opened issue #42